### ISSUE TITLE
v4: incorrect rules for SASS / CSS loaders in projects that have Next in a subdirectory

### ISSUE THREAD

**Issue by anmonteiro** [martpie/next-transpile-modules#94]
- [x] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE

**Are you trying to transpile a local package or an npm package?**
local package

**Describe the bug**
Next.js migrated to using Webpack 5 loader config with `issuer.and`, `or` and `not` in https://github.com/vercel/next.js/pull/14268/files#diff-0df73c8e15d5a4ab2409907c614af8ebL119-L122.

the value for `and` is the root directory for Next.js, but that may not be the directory where `node_modules` is located. My project, for example, has the following structure:

```
.
├── frontend
│   ├── pages
│   │   ├── ....
├── node_modules
├── package.json
...
```

All my Next scripts are e.g. `next build ./frontend`, they pass in the subdirectory.

Now, on to the real issue: `next-transpile-modules` added support for Webpack 5 in #91, however the generated rules are now:

```js
      issuer: {
        and: ['/path/to/project-root/frontend'],
        or: [ /* INSERTED BY `next-transpile-modules` */ ],
      },
```

So, in the case where folks have their Next.js project in subdirectories and `node_modules` one directory above, the strict `and` rule doesn't allow webpack to see the `includes`.

**To Reproduce**
Have a Next.js project in a subdirectory and add `with-transpile-modules` with literally any module that has CSS / SASS to the project.

**Expected behavior**
Project compiles without errors.

**Setup**

- Next.js version: 9.5
- `next-transpile-modules` version: 4.0
- Node.js version: v14.6
- `npm`/`yarn` version: Yarn 1.22.4
- Operating System: Linux

**Additional context**

**Comment by martpie** [comment]
Hello👋

I'll try to have a look when I can. The best you can do to help us would be to submit a failing test :)

**Comment by anmonteiro** [comment]
This is the mitigation I added to my project:

```js
      const nextCssLoaders = config.module.rules.find(
        (rule) => typeof rule.oneOf === 'object'
      );
      if (nextCssLoaders) {
        nextCssLoaders.oneOf.forEach((loader) => {
          if (
            loader.issuer != null &&
            loader.issuer.and != null &&
            loader.sideEffects === false
          ) {
            loader.issuer.and = [__dirname];
          }
        });
      }```

**Comment by martpie** [comment]
I have a potential fix available here: https://github.com/martpie/next-transpile-modules/pull/96

Would you be able to test it before I release it to tell me if it solves your problem?

**Comment by justincy** [comment]
@anmonteiro where did you add that fix?

**Comment by anmonteiro** [comment]
@justincy the full snippet (inside `next.config.js`):

```js
    webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
      https: const nextCssLoaders = config.module.rules.find(
        (rule) => typeof rule.oneOf === 'object'
      );
      if (nextCssLoaders) {
        nextCssLoaders.oneOf.forEach((loader) => {
          if (
            loader.issuer != null &&
            loader.issuer.and != null &&
            loader.sideEffects === false
          ) {
            loader.issuer.and = [__dirname];
          }
        });
      }


      return config;
    },
```

**Comment by martpie** [comment]
Should hopefully be fixed in 4.0.1, please share feedback if it works (or not)!

**Comment by anmonteiro** [comment]
4.0.1 fixes this issue for me. Thanks!
